controller/util/cf_v2_client_test.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+
+	model "github.com/kubernetes-incubator/service-catalog/model/service_broker"
+	scmodel "github.com/kubernetes-incubator/service-catalog/model/service_controller"
+)
+
+func TestUpdateServiceInstance(t *testing.T) {
+	cli := CreateCFV2BrokerClient(&scmodel.ServiceBroker{})
+
+	_, err := cli.UpdateServiceInstance("foo", &model.ServiceInstanceRequest{})
+	if err == nil {
+		t.Fatalf("Expected not implemented")
+	}
+	if err.Error() != "Not implemented" {
+		t.Errorf("Expected not implemented, got %v", err)
+	}
+
+	// TODO: test against fake/test broker.
```

Comment:
Instead of a fake client, it would IMO be better to use a [`(net/http/httptest).Server`](https://godoc.org/net/http/httptest#Server) and the real client. FWIW. @pmorie if you can point me to the issue you've created, I'll add this information there.